hypothesis-python/src/hypothesis/searchstrategy/strategies.py

```diff
@@ -435,12 +435,16 @@ def __init__(self, strategies, bias=None):
         self.__element_strategies = None
         self.bias = bias
         self.__in_branches = False
-        if bias is not None:
-            assert 0 < bias < 1
-            self.sampler = cu.Sampler(
-                [bias ** i for i in range(len(strategies))])
-        else:
-            self.sampler = None
+        self.__sampler = None
+
+    @property
+    def sampler(self):
```

Comment:
I wonder how much we actually benefit from all this laziness. I don't even remember why we do it this way. Maybe we should just assign these in the constructor?

(Thinking out loud, you don't need to do anything about this comment if you don't want to)

Reply:
Seems plausible, but I'll leave it for the new issue - no need to tempt Murphy here!